projs/shadow/shadow-engine/shadow-entity/Entities/FirstPersonCamera.cpp

```diff
@@ -0,0 +1,96 @@
+#include "shpch.h"
+#include "FirstPersonCamera.h"
+
+#include "ShadowInput/ShadowAction.h"
+#include "ShadowInput/Bindings/Binding1D.h"
+#include "ShadowInput/Bindings/KeyboardBinding.h"
+#include "ShadowInput/Bindings/MousePositionBinding.h"
+#include "ShadowInput/Bindings/MouseButtonBinding.h"
+#include "ShadowInput/Bindings/BindingAND.h"
+
+#include "EntitySystem/EntityManager.h"
+
+#include "EntitySystem/Scene.h"
+
+#include <iostream>
+
+#include <SDL.h>
+
+void FirstPersonCamera::Build()
+{
+	cam = AddInternalChildEntity<Camera>();
+	cam->name = "Camera";
+}
+
+void FirstPersonCamera::Init()
+{
+	//cam = scene->AddEntity<Camera>();
+
+	cam->SetCameraType(CameraType::Perspective);
+	cam->transform.SetPosition({ 0,0.0f,0.0f });
+
+
+
+
+	forward = new ShadowEngine::InputSystem::ShadowAction<float>(
+		"Forward",
+		new ShadowEngine::InputSystem::Binding1D(
```

Comment:
This binding system looks neat. We should tear this out and implement it under the internal event system.

Reply:
> The Entities folder contains old built-in entities but they are currently not included in the project. Needs sorting and updating to the new renderer stuff.

No.
That is my old input system inspired by Unity's new input system. While the interface looks fun you would hate the implementation as it is template hell even by my standards. Something similar is planned for the input/action handling, but that needs time.